internal/v2/s2av2.go

```diff
@@ -0,0 +1,99 @@
+// Package s2av2 provides the S2Av2 transport credentials used by a gRPC
+// application.
+package s2av2
+
+import (
+	"errors"
+	"context"
+	"net"
+	"crypto/tls"
+
+	"google.golang.org/grpc/credentials"
+	"github.com/google/s2a-go/internal/v2/tls_config_store"
+)
+
+const (
+	s2aSecurityProtocol = "s2av2"
+)
+
+type s2av2TransportCreds struct {
+	info     *credentials.ProtocolInfo
+	isClient bool
+	serverName string
+}
+
+// NewClientCreds returns a client-side transport credentials object that uses
+// the S2Av2 to establish a secure connection with a server.
+func NewClientCreds(serverName string) (credentials.TransportCredentials, error) {
```

Comment:
nit: I think we should omit the `serverName` parameter and leave it uninitialized (or explicitly initialize it to the empty string).

Reply:
Will do! So when a client wants to set the serverName, they will have to use OverrideServerName?

Reply:
Correct.